openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -162,3 +176,41 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 	}
 	return route.Status.Ingress[0].Host
 }
+
+// checkMinimumVersion checks if the version in the arg meets the requirement or not.
+// It is similar logic with CheckMinimumVersion() in knative.dev/pkg/version.
+func checkMinimumVersion(versioner discovery.ServerVersionInterface, version string) error {
+	v, err := versioner.ServerVersion()
+	if err != nil {
+		return err
+	}
+	currentVersion, err := semver.Make(normalizeVersion(v.GitVersion))
+	if err != nil {
+		return err
+	}
+
+	minimumVersion, err := semver.Make(normalizeVersion(version))
+	if err != nil {
+		return err
+	}
+
+	// If no specific pre-release requirement is set, we default to "-0" to always allow
+	// pre-release versions of the same Major.Minor.Patch version.
+	if len(minimumVersion.Pre) == 0 {
+		minimumVersion.Pre = []semver.PRVersion{{VersionNum: 0}}
```

Comment:
I suppose we should do the IsNum thing we did upstream here as well?

Reply:
Thank you! Yes it should be added. Updated.